darts/models/forecasting/forecasting_model.py

```diff
@@ -382,7 +385,21 @@ def historical_forecasts(
 
         # iterate and forecast
         for pred_time in iterator:
-            train = series.drop_after(pred_time)  # build the training series
+            # build the training series
+            train = series.drop_after(pred_time)
+            if train_length and not isinstance(train_length, int):
```

Comment:
These checks should be done earlier in the method (as opposed to in every iteration of the loop)